File: project-c-cosminpac04/motorcycleApp/network/ClientNotifier.cs
using System;
using System.IO;
using System.Net.Sockets;

namespace motorcycleApp.network
{
    public class ClientNotifier
    {
        private readonly TcpClient _clientSocket;  
        private readonly NetworkStream _networkStream;

        public ClientNotifier(TcpClient clientSocket)
        {
            _clientSocket = clientSocket;
            _networkStream = _clientSocket.GetStream(); 
        }

        public void SendUpdateRequest()
        {
            try
            {
                var writer = new StreamWriter(_networkStream);
                writer.WriteLine("REQUEST_UPDATE");
                writer.Flush();
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error in sending update request: {ex.Message}");
            }
        }
    }
}

File: project-java-cosminpac04/motorcycleContest/src/main/java/motorcycle/network/ClientNotifier.java
package motorcycle.network;

import java.io.PrintWriter;
import java.net.Socket;

public class ClientNotifier {

    private final Socket socket;

    public ClientNotifier(Socket socket) {
        this.socket = socket;
    }

    public void sendUpdateRequest() {
        try {
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            out.println("REQUEST_UPDATE");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
